mail_operating_unit/models/mail_mail_ext.py

```diff
@@ -17,25 +17,25 @@ def create(self, vals_list):
         mails = super(MailMailExt, self).create(vals_list)
 
         for mail in mails:
-            if mail.mail_message_id.model and \
-                    mail.mail_message_id.res_id:
-
+            if mail.mail_message_id.model and mail.mail_message_id.res_id:
                 model = mail.mail_message_id.model
                 res_id = mail.mail_message_id.res_id
                 record = self.env[model].browse(res_id)
-
+                operating_unit = False
                 if getattr(record, 'operating_unit_id', False):
-                    if record.operating_unit_id.catchall_alias and \
-                            record.operating_unit_id.catchall_domain:
-                        sender_email_address = \
-                            record.operating_unit_id.catchall_alias + '@' + \
-                            record.operating_unit_id.catchall_domain
-                        email_name = record.operating_unit_id.partner_id.name
-                        mail.email_from = \
-                            formataddr((email_name, sender_email_address))
-                        mail.reply_to = mail.email_from
-                    if record.operating_unit_id.outgoing_mail_server_id:
-                        mail.mail_server_id =\
-                            record.operating_unit_id.outgoing_mail_server_id.id
+                    operating_unit = record.operating_unit_id
+                elif self.env.context.get('uid', False):
+                    user = self.env['res.users'].browse(self.env.context.get('uid'))
+                    operating_unit = user.operating_unit_for_mails_id
+
+                if operating_unit and operating_unit.catchall_alias and operating_unit.catchall_domain:
+                    sender_email_address = (operating_unit.catchall_alias + '@' +
+                                            operating_unit.catchall_domain)
+                    email_name = operating_unit.partner_id.name
+                    mail.email_from = formataddr((email_name, sender_email_address))
+                    mail.reply_to = mail.email_from
+
+                    if operating_unit.outgoing_mail_server_id:
+                        mail.mail_server_id = operating_unit.outgoing_mail_server_id.id
```

Comment:
When possible, avoid changing the layout even if you don't like it, because it makes difficult to know what changed.

Reply:
we had to rewrite the whole part. for this you need to view the file itself, that's how I do it.